code/_lighting/SimpleLight.dm

```diff
@@ -346,47 +344,48 @@
 	switch(direct)
 		if (NORTH)
 			vx = 0
-			vy = 1
+			vy = 32
 		if (NORTHEAST)
-			vx = 0.7071
-			vy = 0.7071
+			vx = 22.627417 //sqrt(2)/2 * 32
+			vy = 22.627417
 		if (EAST)
-			vx = 1
+			vx = 32
 			vy = 0
 		if (SOUTHEAST)
-			vx = 0.7071
-			vy = -0.7071
+			vx = 22.627417
+			vy = -22.627417
 		if (SOUTH)
 			vx = 0
-			vy = -1
+			vy = -32
 		if (SOUTHWEST)
-			vx = -0.7071
-			vy = -0.7071
+			vx = -22.627417
+			vy = -22.627417
 		if (WEST)
-			vx = -1
+			vx = -32
 			vy = 0
 		if (NORTHWEST)
-			vx = -0.7071
-			vy = 0.7071
+			vx = -22.627417
+			vy = 22.627417
 
 	var/turf/T = get_steps(src, direct, 5)
-	if (!src || !T)
+	if (!T)
 		return
 	var/turf/TT = getlineopaqueblocked(src,T)
 	var/dist = GET_DIST(src,TT)-1
 
-	for (var/obj/overlay/simple_light/medium/directional/light as anything in src.mdir_lights)
+	for (var/atom/movable/light/simple_light/medium/directional/light as anything in src.mdir_lights)
 		if(light.icon_state == "medium_center" && light.dist == 0)
 			src:vis_contents += light
 			continue
 
 		////light.pixel_x = (vx * min(dist,light.dist) * 32) - 32
 		//light.pixel_y = (vy * min(dist,light.dist) * 32) - 32
 
-		animate(light,pixel_x = ((vx * min(dist,light.dist) * 32) - 32), pixel_y = ((vy * min(dist,light.dist) * 32) - 32), time = 1, easing = EASE_IN)
```

Comment:
que?

Reply:
microoptimisation, i moved the * 32 to the vx and vy. i should probably make it 0.7071 * 32 in the switch so itd be clearer and still constfold